### PR TITLE
feat: White bar similarity based on variance

### PR DESCRIPTION
**Pull request by bartoszgorka**
```bash
WYNIKI:
set0 = [4. 0. 1. 1. 0.] score = 4.583333333333334 [0sec]
set1 = [17.  1.  0.  0.  0.] score = 17.767857142857142 [0sec]
set2 = [11.  3.  1.  1.  0.] score = 13.393589743589745 [0sec]
set3 = [17.  0.  1.  0.  0.] score = 17.549242424242422 [0sec]
set4 = [16.  0.  0.  0.  1.] score = 16.578968253968252 [0sec]
set5 = [108.   9.   3.   3.   3.] score = 117.04444349291713 [1sec]
set6 = [102.   2.   4.   2.   2.] score = 107.72693586057187 [1sec]
set7 = [2. 1. 1. 1. 1.] score = 4.710989010989011 [0sec]
set8 = [9. 4. 0. 1. 0.] score = 13.564156839575531 [0sec]
----
[286.  20.  11.   9.   7.] score = 312.91951610204444 [5sec]
```

### CODE REVIEW

File: project_1/piro1/image.py [hunk 1]
```diff
@@ -47,3 +48,10 @@ def rotate_with_scale(img):
 
         # Return rotated, reversed and scaled image
         return image
+
+    def calculate_white_bars(self):
+        vector = np.empty(shape=300, dtype=np.uint16)
```
Comment (hejmsdz) [inline project_1/piro1/image.py]: It would be nice to extract these fixed dimensions into some constant.

File: project_1/piro1/matcher.py [hunk 2]
```diff
@@ -19,15 +21,14 @@ def match(self):
...
                 self.matching[i, j] += score
                 self.matching[j, i] += score
-    
+
     def best_matches(self):
         for i in range(self.n):
             row = self.matching[i, :]
-            yield [j for j in np.argsort(-row) if j != i and row[j] > 0]
-    
+            yield [j for j in np.argsort(-row) if j != i and row[j] >= 0]
```
Comment (hejmsdz) [inline project_1/piro1/matcher.py]: All match values should be between 0 and 1, so if you're dropping this constraint, you may as well remove the whole `and row[j] >= 0`.

File: project_1/piro1/image.py (hunk 1 above)
Comment (bartoszgorka) [reply project_1/piro1/image.py]: I agree with you, changed 👍

**Comment by bartoszgorka** [comment]
@hejmsdz after changes:
```bash
WYNIKI:
set0 = [6. 0. 0. 0. 0.] score = 6.0 [0sec]
set1 = [20.  0.  0.  0.  0.] score = 20.0 [0sec]
set2 = [18.  1.  0.  0.  0.] score = 18.55263157894737 [0sec]
set3 = [20.  0.  0.  0.  0.] score = 20.0 [0sec]
set4 = [20.  0.  0.  0.  0.] score = 20.0 [0sec]
set5 = [177.   5.   2.   0.   3.] score = 181.6732772673949 [1sec]
set6 = [154.   4.   0.   2.   0.] score = 157.76784238175557 [1sec]
set7 = [14.  2.  3.  1.  0.] score = 16.25 [0sec]
set8 = [18.  6.  4.  5.  5.] score = 28.666800684048038 [0sec]
----
[447.  18.   9.   8.   8.] score = 468.9105519121459 [5sec]
```

**Pull request merged**